### ISSUE TITLE
Allow scenarios to specify starting weather.

### ISSUE THREAD

**Issue by DoctorBoomstick** [CleverRaven/Cataclysm-DDA#63444]
### Is your feature request related to a problem? Please describe.

When making a new scenario, there are a number of things one may do to make it interesting, such as customizing the time, spawn location, and applying effects. However, one of the elements beyond the scope of scenarios is specifying a particular type of weather that the game starts with. This evidently holds through for EOCs from what I’ve been told, and is, in my personal view, a slight shame as determining the weather can lead to some interesting scenario ambiences, as well as opening the door to new ideas.

### Solution you would like.

I’d suggest allowing weather effects to be prompted through EOCs to leave room for interesting ingame uses as well as for utility in scenarios, or simply allowing the type of desired weather to be specified in the scenario json.

**Comment by ZhilkinSerg** [comment]
I think you can use `SCENARIO_SPECIFIC` EOC and `next_weather` effect to force specific weather that you want in your scenario. You can see example in https://github.com/CleverRaven/Cataclysm-DDA/blob/0f4eccae1290c34a4ee425d98ba850bc5cb5b29e/data/mods/desert_region/weather/weather_eoc.json or https://github.com/CleverRaven/Cataclysm-DDA/blob/2ee0a718b55004f7f30fae29c4fb2265fd930f72/data/json/portal_storm_effect_on_condition.json

**Comment by DoctorBoomstick** [comment]
From what I’ve been told, it looks like both the “cause_portal_storm” and “cause_dust_storm” EOCs hinge on all their starting variables being set by the EOC in question. For weather to similarly be caused, such as a thunder storm, an EOC would have to set up precipitation and other weather values to prompt the start of a weather change: something that can’t be done at the present.

**Comment by GuardianDll** [comment]
Right, as you may expect from its name, `next_weather` do not force the game to change the weather to some picked, it force the game to see what another weather it can pick, and pick it. Both dust storm and portal storm has no another conditions except few variables, when to start, for example, rainstorm, you need to have temperature more than 33, and wind speed more than 15 (https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/weather_type.json#L153)
Without a way to force or avoid this checks, the game will simply find it can't apply them, and casually reject them.

**Comment by Ramza13** [comment]
Ok so its awkward but you could just add an or condition to the weather you want to force with a variable.  So for example rainstorm is either temp > 33 and wind >15 or is_rain = 1.  Then set the variable for as long as you like and boom.  A bit complicated but doable.

**Comment by JonathanLochridge** [comment]
I agree this would be cool to have. After reading this I already have a couple ideas of how I could use this if we had it. 
Most notably making a winter start that makes you start in the wilderness in the middle of a blizzard. So, that finding shelter or a source of warmth is essential to not die to the cold. (Although, spawning with someone with warmer equipment could give you a little bit of a margin.)
